### PR TITLE
Update the Ginkgo package to release 1.0.0

### PR DESCRIPTION
**Pull request by tcojean**
A first spack package version of the [Ginkgo software](https://github.com/ginkgo-project/ginkgo/) was recently added by @davydden. Since then, we have fixed several problems which were present at the time, and we have released the first version of our software (1.0.0). 
In this PR, I hope to improve the spack Ginkgo package to reflect these changes.

Details:
+ Fix a minor typo in the deal.II package.
+ Version 1.0.0 was released and the master branch tracks
   the latest release.
+ Multiple issues with MacOS have been fixed.
+ The static patch is now unneeded.
+ Add support for building Ginkgo with the full Block Jacobi
   performance (compilation takes a long time, so it's off by
   default).
+ Do not build any documentation, benchmarks, examples
   and tests as they are not used at installation time.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ginkgo/package.py [hunk 1]
```diff
@@ -11,16 +11,16 @@ class Ginkgo(CMakePackage, CudaPackage):
     """High-performance linear algebra library for manycore systems,
     with a focus on sparse solution of linear systems."""
 
-    homepage = "https://github.com/ginkgo-project/ginkgo"
+    homepage = "https://ginkgo-project.github.io/"
     url      = "https://github.com/ginkgo-project/ginkgo.git"
     git      = "https://github.com/ginkgo-project/ginkgo.git"
 
     version('develop', branch='develop')
+    version('latest', branch='master')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ginkgo/package.py]: Please don’t call it latest, but ‘master’ instead, version comparison will soon work fine for this case as well.

Comment (tcojean) [reply var/spack/repos/builtin/packages/ginkgo/package.py]: Alright, I put `master` here.

File: var/spack/repos/builtin/packages/ginkgo/package.py [hunk 2]
```diff
@@ -29,15 +29,18 @@ class Ginkgo(CMakePackage, CudaPackage):
     depends_on('cmake@3.9:', type='build')
     depends_on('cuda@9:',    when='+cuda')
```
Comment (pratikvn) [inline var/spack/repos/builtin/packages/ginkgo/package.py]: Will the CudaArchitectureSelector work properly here ? Also, I don't think there are any tests which have `cuda` as a spec. Maybe I missed something ?

**Comment by tcojean** [comment]
@pratikvn sure, why wouldn't the CAS not work? It will use the default value for building, either all architectures or the specific detected architecture.

The `cuda` flag is here to control whether or not the user wants to build Ginkgo with `cuda` support. If `cuda` is passed to `spack` by the user, then the package passes `-DGINKGO_BUILD_CUDA=ON` as a build-time option, that is all it's here for.

**Pull request merged**